components/raftstore/src/store/fsm/store.rs

```diff
@@ -3130,13 +3133,14 @@ impl<EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'_, EK, ER, T>
         self.register_compact_lock_cf_tick();
     }
 
-    fn on_wake_up_regions(&self, abnormal_stores: Vec<u64>) {
+    fn on_wake_up_regions(&self, abnormal_stores: Vec<u64>, region_ids: Vec<u64>) {
         info!("try to wake up all hibernated regions in this store";
             "to_all" => abnormal_stores.is_empty());
         let store_id = self.ctx.store_id();
         let meta = self.ctx.store_meta.lock().unwrap();
 
-        for (region_id, region) in &meta.regions {
+        for region_id in region_ids {
+            let region = &meta.regions[&region_id];
```

Comment:
I've realized that a panic may occur if a region is removed after the PD worker sends the region_ids.